usb.go

```diff
@@ -210,6 +227,27 @@ func (c *Context) OpenDevices(opener func(desc *DeviceDesc) bool) ([]*Device, er
 	return ret, reterr
 }
 
+func (c *Context) OpenDeviceWithFileDescriptor(fd uintptr) (*Device, error) {
+	handle, err := c.libusb.wrapSysDevice(c.ctx, fd)
+	if err != nil {
+		return nil, fmt.Errorf("Error opening device from file descriptor: %d, %s", fd, err.Error())
+	}
+
+	dev := c.libusb.getDevice(handle)
+
+	desc, err := c.libusb.getDeviceDesc(dev)
+	if err != nil {
+		return nil, err
```

Comment:
if you want to annotate an error somewhere, here it would be appropriate perhaps:
fmt.Errorf("device was opened, but get device descriptor failed: %v")